test/geometry/LinearPropagator.test.cc

```diff
@@ -85,56 +86,57 @@ TEST_F(LinearPropagatorHostTest, track_line)
         geo = {{-10, 10, 10}, {1, 0, 0}};
         EXPECT_EQ(VolumeId{0}, geo.volume_id()); // Shape2 center
 
-        geo.find_next_step();
-        EXPECT_SOFT_EQ(5, geo.next_step());
-        propagate();
+        auto step = propagate(1.e10); // very large proposed step
+        EXPECT_SOFT_EQ(5, step.distance);
+        EXPECT_EQ(VolumeId{1}, step.volume); // Shape2 -> Shape1
         EXPECT_SOFT_EQ(-5, geo.pos()[0]);
-        EXPECT_EQ(VolumeId{1}, geo.volume_id()); // Shape2 -> Shape1
 
-        geo.find_next_step();
-        EXPECT_SOFT_EQ(1.0, geo.next_step());
-        propagate();
-        EXPECT_EQ(VolumeId{3}, geo.volume_id()); // Shape1 -> Envelope
+        step = propagate(1.e10);
+        EXPECT_SOFT_EQ(1, step.distance);
+        EXPECT_EQ(VolumeId{3}, step.volume); // Shape1 -> Envelope
         EXPECT_EQ(false, geo.is_outside());
 
-        geo.find_next_step();
-        EXPECT_SOFT_EQ(1.0, geo.next_step());
-        propagate();
+        step = propagate();
+        EXPECT_SOFT_EQ(1, step.distance);
         EXPECT_EQ(false, geo.is_outside()); // leaving World
     }
 
+    try
     {
         // Track from outside edge fails
+        CELER_LOG(info) << "Init a track with a pointer outside work "
+                           "volume...";
         geo = {{24, 0, 0}, {-1, 0, 0}};
         EXPECT_EQ(true, geo.is_outside());
     }
+    catch (const std::exception& e)
+    {
+    }
```

Comment:
Same comment here. Is the comment above the new log statement accurate?

Reply:
Yes, it is accurate.  The call to `is_outside()` throws an exception, and the test fails. 
I tried to catch the exception, to avoid the test failure.  What do you suggest instead?

Reply:
I used EXPECT_THROW() instead of the try-catch block.